FormalConjectures/Wikipedia/CongruentNumber.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Congruent Number
+
+A natural number $n$ is called a congruent number if there exists a right triangle with rational
+sides $a$, $b$, and hypotenuse $c$ such that the area of the triangle is $\frac{1}{2}ab = n$.
+
+*References:*
+- [Wikipedia (Congruent number)](https://en.wikipedia.org/wiki/Congruent_number)
+- [Wikipedia (Tunnell's theorem)](https://en.wikipedia.org/wiki/Tunnell%27s_theorem)
+- [Keith Conrad's note](https://kconrad.math.uconn.edu/blurbs/ugradnumthy/congnumber.pdf)
+-/
+
+
+def congruentNumber (n : ℕ) : Prop :=
+  ∃ (a b c : ℚ), a ^ 2 + b ^ 2 = c ^ 2 ∧ n = (2⁻¹ : ℚ) * a * b
+
+/- 1 is not a congruent number. -/
+@[category test]
+example : ¬ congruentNumber 1 := by
+  sorry
+
+/- 5, 6, 7, and 157 are congruent numbers. -/
+@[category test]
+example : congruentNumber 5 := by
+  use 3 / 2, 20 / 3, 41 / 6
+  norm_num
+
+@[category test]
+example : congruentNumber 6 := by
+  use 3, 4, 5
+  norm_num
+
+@[category test]
+example : congruentNumber 7 := by
+  use 35 / 12, 24 / 5, 337 / 60
+  norm_num
+
+/- Zagier's example -/
+@[category test]
+example : congruentNumber 157 := by
+  use 411340519227716149383203 / 21666555693714761309610,
+    6803298487826435051217540 / 411340519227716149383203,
+    224403517704336969924557513090674863160948472041 /
+      8912332268928859588025535178967163570016480830
+  norm_num
+
+
+/- Characterize congruent numbers. -/
+theorem congruent_number_problem (n : ℕ+) : congruentNumber n ↔ answer(sorry) := by
```

Comment:
```suggestion
/- Characterize congruent numbers. -/
@[category research solved, AMS 11]
theorem congruent_number_problem (n : ℕ+) : congruentNumber n ↔ answer(sorry) := by
```

I'm not entirely convinced that this request for characterization is a good idea: is there reason to expect there is a simple characterization?